lightning/src/ln/channelmanager.rs

```diff
@@ -4375,10 +4458,16 @@ impl_writeable_tlv_based!(HTLCPreviousHopData, {
 	(6, incoming_packet_shared_secret, required)
 });
 
+impl_writeable_tlv_based_enum!(ClaimableType, ;
+	(0, Invoice),
+	(1, Spontaneous),
+);
+
 impl_writeable_tlv_based!(ClaimableHTLC, {
 	(0, prev_hop, required),
 	(2, value, required),
-	(4, payment_data, required),
+	// XXX what to do here?
+	(4, claim_type, required),
```

Comment:
Hmm, right, so to make this compatible we need to pull the `impl_writeable_tlv_based` apart and implement manually - if `ClaimableType` is `Invoice` we'll write the 4-type `payment_data` and if its `Spontaneous` we'll write an 8-type preimage. Its not ideal and we could just cheat here if we want, but IMO its not a ton of work here.

Reply:
I *think* I did this